ci-bench/src/main.rs

```diff
@@ -183,6 +189,70 @@ fn main() -> anyhow::Result<()> {
             mem::forget(stdin);
             mem::forget(stdout);
         }
+        Command::Walltime {
+            iterations_per_scenario,
+        } => {
+            let mut timings = vec![Vec::with_capacity(iterations_per_scenario); benchmarks.len()];
+            for _ in 0..iterations_per_scenario {
+                for (i, bench) in benchmarks.iter().enumerate() {
+                    let start = Instant::now();
+
+                    let resumption_kind = black_box(bench.kind.resumption_kind());
+                    let params = black_box(&bench.params);
```

Comment:
How come these get a `black_box` hint? I would expect `run_bench` would benefit from that but I'm not sure I understand why these two invocations are used. 

Can you add a comment to help me understand?

Reply:
I must say I find it difficult to get my thoughts straight on this one, but I wrote a comment that hopefully clarifies things. Otherwise, let me know!

Reply:
I think I understand better now, thank you.

Apologies, but I didn't notice that there were equivalent `black_box!` invocations earlier in the code on L156 and L157. Maybe it would be better to put the comment there and then reference it here?

Reply:
The funny thing is... I don't think we actually need the black box on L156 and L157, because in that case the server and the client run in different processes, so each side of the connection has no compile-time information about the other side. Just pushed a commit to remove them.